sdk/internal/recording/recording.go

```diff
@@ -433,3 +438,221 @@ var modeMap = map[RecordMode]recorder.Mode{
 	Live:     recorder.ModeDisabled,
 	Playback: recorder.ModeReplaying,
 }
+
+var recordMode, _ = os.LookupEnv("AZURE_RECORD_MODE")
+
+const (
+	modeRecording      = "record"
+	modePlayback       = "playback"
+	baseProxyURLSecure = "localhost:5001"
+	baseProxyURL       = "localhost:5000"
+	IdHeader           = "x-recording-id"
+	ModeHeader         = "x-recording-mode"
+	UpstreamUriHeader  = "x-recording-upstream-base-uri"
+)
+
+var recordingId string
+
+var tr = &http.Transport{
+	TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+}
+var client = http.Client{
+	Transport: tr,
+}
+
+type RecordingOptions struct {
+	UseHTTPS bool
+	Host     string
+	Scheme   string
+}
+
+func resetRecordingId() {
```

Comment:
If I were to run tests in parallel (which I think is a good option for unit tests) would that break what you're doing?

Reply:
It probably would. I'm not sure how to handle parallel tests right now and haven't tested it but I will add it as a todo. I agree it's something we should be able to handle.